### DISCUSSION TITLE
How I use my old client secret(not keycloak server) in keycloak.

### DISCUSSION THREAD

**Discussion by asd332689912** [keycloak/keycloak#22712] [Q&A]
I have an SSO system with many clients on it, now I want to migrate the above client to keycloak and use the former secret, what should I do?

I have started keycloak using mysql and written the data, but modifying the mysql data directly is of no use.

**Comment by christian-2** [comment]
Do you mean secrets as in OAuth 2.0/OIDC [client secret](https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1)?

If so, you should be able to migrate your secrets into Keycloak e.g. with `kcadm.sh update -r $realm clients/$id -s secret=$secret`.

**Reply by asd332689912** [reply]
s there any way to directly read secret data in mysql? I can directly modify secret in batches in mysql。
